### ISSUE TITLE
First group member doesn't send messages

### ISSUE THREAD

**Issue by taylorflatt** [taylorflatt/go-chat#1]
```
2017/03/17 19:25:12 [addClient]: Registered client user1
2017/03/17 19:25:13 [addGroup]: Added group group1
2017/03/17 19:25:13 [addClientToGroup] Added user1 to group1
2017/03/17 19:25:20 [addClient]: Registered client user2
2017/03/17 19:25:21 [GetGroupList]: Returned list of current groups
2017/03/17 19:25:21 [group1]
2017/03/17 19:25:23 [JoinGroup] Attempting to add user2 to groupgroup1
2017/03/17 19:25:23 [addClientToGroup] Added user2 to group1
2017/03/17 19:25:25 [RouteChat]: Client user2 sent group1 a message: hello
2017/03/17 19:25:28 [RouteChat]: Client user1 sent  a message: how are you?
2017/03/17 19:25:33 Client user2 sent group1 a message: I am well

2017/03/17 19:25:41 Client user1 sent  a message: what?

2017/03/17 19:25:57 [addClient]: Registered client user3
2017/03/17 19:25:58 [GetGroupList]: Returned list of current groups
2017/03/17 19:25:58 [group1]
2017/03/17 19:26:00 [JoinGroup] Attempting to add user3 to groupgroup1
2017/03/17 19:26:00 [addClientToGroup] Added user3 to group1
2017/03/17 19:26:05 [RouteChat]: Client user3 sent group1 a message: hi, I am user3
2017/03/17 19:26:15 Client user3 sent group1 a message: 2
2017/03/17 19:26:22 Client user3 sent group1 a message: Hi hello
2017/03/17 19:26:29 Client user3 sent group1 a message: hi
```

Every other message gets sent by any user (not user1).

The first user's messages won't ever get sent and apparently the group name isn't added. Although, it is added to the groupList (i.e. it shows user1, user2, and user3 when displayed).

Note that the last three messages are logged from the listToClientMethod whereas the 4th from the last message is logged from the RouteChat.

**Comment by taylorflatt** [comment]
The main problem is two-fold:

~~1. The first client doesn't send/receive any messages (no group is displayed).~~
2. The other clients that join the group, they only receive every other message of other clients.

**Comment by taylorflatt** [comment]
Fixed the first issue with [new commit](https://github.com/taylorflatt/go-chat/commit/a2509d965e52a68aa64c0a321bca50c955201e13) (line 145 of the client). Essentially what happened was that a variable with the same name as the global gName variable was being assigned only when a group was created. 

The other problem is still on-going.

**Comment by taylorflatt** [comment]
It appears as if the messages are getting split between the other clients in the group.

Note the following messages from Client3.

Client1:
```
hi1
hi1a
hi1b
hi1c
user2> hi2a
user2> hi2b
user2> hi2c


user3> hi3b
user3> hi3d
user3> hi3f
```

Client2:
```
hi
user1> hi1a
user1> hi1b
user1> hi1c
hi2a
hi2b
hi2c


user3> hi3a
user3> hi3c
user3> hi3e
user3> hi3g
```

Client3:
```
hi3
hi3a
hi3b
hi3c
hi3d
hi3e
hi3f
hi3g
```

As a side note: The first message for every client is also not sent to everyone. That problem has persisted from the first iteration.

**Comment by taylorflatt** [comment]
This has been fixed in #3.
